agents/ahv/fence_ahv.py

```diff
@@ -0,0 +1,608 @@
+
+        logging.info("Powered %s VM, %s successfully",
+                     power_state.lower(), vm_uuid)
+
+    def power_cycle_vm(self, vm_name=None, vm_uuid=None, timeout=None):
+        resp = None
+        status = None
+
+        if not timeout:
+            timeout = MIN_TIMEOUT
+
+        if not vm_name and not vm_uuid:
+            logging.error("Require at least one of VM name or VM UUID")
+            raise AHVFenceAgentException("No arguments provided")
+
+        if not vm_uuid:
+            vm_uuid = self._get_vm_uuid(vm_name)
+
+        resp = self._power_cycle_vm(vm_uuid)
+        task_id = resp.json()['data']['extId']
+
+        try:
+            self._wait_for_task(task_id, timeout)
+        except AHVFenceAgentException as err:
+            logging.error("Failed to power-cycle VM %s", vm_uuid)
+            logging.error("VM power-cycle task failed with status, %s", status)
+            raise AHVFenceAgentException from err
+        except TaskTimedOutException as err:
+            logging.error("Timed out power-cycling VM %s", vm_uuid)
+            raise TaskTimedOutException from err
+
+
+        logging.info("Power-cycled VM, %s", vm_uuid)
+
+
+def connect(options):
+    host = options["--ip"]
+    username = options["--username"]
+    password = options["--password"]
+    verify_ssl = True
+    disable_warnings = False
+
+    if "--ssl-insecure" in options:
+        verify_ssl = False
+        disable_warnings = True
+
+    client = NutanixV4Client(host, username, password,
+                             verify_ssl, disable_warnings)
+
+    try:
+        client.list_vms(limit=1)
+    except AHVFenceAgentException as err:
+        logging.error("Connection to Prism Central Failed")
+        logging.error(err)
+        fail(EC_LOGIN_DENIED)
+
+    return client
+
+def get_list(client, options):
+    vm_list = None
+    limit = None
+    filter_str =  None
+
+    if "--filter" in options:
+        filter_str = options["--filter"]
+
+    if "--limit" in options:
+        limit = options["limit"]
+
+    try:
+        vm_list = client.list_vms(filter_str, limit)
+    except AHVFenceAgentException as err:
+        logging.error("Failed to list VMs")
+        logging.error(err)
+        fail(EC_GENERIC_ERROR)
+
+    return vm_list
+
+def get_power_status(client, options):
+    vmid = None
+    name = None
+    power_state = None
+
+    if "--uuid" in options:
+        vmid = options["--uuid"]
+    else:
+        name = options["--plug"]
+
+    if not vmid and not name:
+        logging.error("Need VM name or VM UUID for power op")
+        fail(EC_GENERIC_ERROR)
+    try:
+        power_state = client.get_power_state(vm_name=name, vm_uuid=vmid)
+    except AHVFenceAgentException:
+        fail(EC_GENERIC_ERROR)
+
+    return power_state
+
+def set_power_status(client, options):
+    vmid = None
+    name = None
+    action = None
+    timeout = None
+
+    if "--action" not in options:
```

Comment:
It shouldnt be able to get here unless --action is on or off.

Reply:
Fixed.